pkg/server/api_v2_grants_test.go

```diff
@@ -0,0 +1,488 @@
+	require.NoError(t, err)
+
+	// Grant different privileges to users.
+	conn.Exec(t, `GRANT CREATE, CONNECT ON DATABASE test_db TO user1`)
+	conn.Exec(t, `GRANT ZONECONFIG, BACKUP ON DATABASE test_db TO user2`)
+	conn.Exec(t, `GRANT ALL ON DATABASE test_db TO user3`)
+	conn.Exec(t, `REVOKE CONNECT ON DATABASE no_access_db FROM public`)
+
+	// Define test cases.
+	testCases := []struct {
+		name           string
+		dbID           int
+		pageSize       int
+		pageNum        int
+		expectedStatus int
+		expectedTotal  int
+		expected       []grantRecord
+		// If specified, only this client will be used.
+		// Otherwise, both the authenticated and admin client will be tested.
+		client *http.Client
+	}{
+		{
+			name:           "Grants on test_db",
+			dbID:           testDBID,
+			pageSize:       0,
+			pageNum:        0,
+			expectedStatus: http.StatusOK,
+			expectedTotal:  8,
+			expected: []grantRecord{
+				{Grantee: "admin", Privilege: "ALL"},
+				{Grantee: "public", Privilege: "CONNECT"},
+				{Grantee: "root", Privilege: "ALL"},
+				{Grantee: "user1", Privilege: "CONNECT"},
+				{Grantee: "user1", Privilege: "CREATE"},
+				{Grantee: "user2", Privilege: "BACKUP"},
+				{Grantee: "user2", Privilege: "ZONECONFIG"},
+				{Grantee: "user3", Privilege: "ALL"},
+			},
+		},
+		{
+			name:           "Grants on test_db limit 5",
+			dbID:           testDBID,
+			pageSize:       5,
+			pageNum:        0,
+			expectedStatus: http.StatusOK,
+			expectedTotal:  8,
+			expected: []grantRecord{
+				{Grantee: "admin", Privilege: "ALL"},
+				{Grantee: "public", Privilege: "CONNECT"},
+				{Grantee: "root", Privilege: "ALL"},
+				{Grantee: "user1", Privilege: "CONNECT"},
+				{Grantee: "user1", Privilege: "CREATE"},
+			},
+		},
+		{
+			name:           "No access to no_access_db (no_access_user)",
+			dbID:           noAccessDBID,
+			pageSize:       10,
+			pageNum:        0,
+			expectedStatus: http.StatusOK,
+			expectedTotal:  0,
+			client:         &authenticatedClient,
+			expected:       []grantRecord{},
+		},
+		{
+			name:           "Admin access to no_access_db",
+			dbID:           noAccessDBID,
+			pageSize:       10,
+			pageNum:        0,
+			expectedStatus: http.StatusOK,
+			client:         &adminClient,
+			expectedTotal:  2,
+			expected: []grantRecord{
+				{Grantee: "admin", Privilege: "ALL"},
+				{Grantee: "root", Privilege: "ALL"},
+			},
+		},
+		{
+			name:           "Page size and number combined on test_db",
+			dbID:           testDBID,
+			pageSize:       2,
+			pageNum:        3,
+			expectedStatus: http.StatusOK,
+			expectedTotal:  8,
+			expected: []grantRecord{
+				{Grantee: "user1", Privilege: "CREATE"},
+				{Grantee: "user2", Privilege: "BACKUP"},
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			var queryParams string
+			if tc.pageSize > 0 {
+				queryParams += fmt.Sprintf("%s=%d", pageSizeKey, tc.pageSize)
```

Comment:
nit: you can make this change remove the if on line 158
```suggestion
				queryParams += fmt.Sprintf("%s=%d&", pageSizeKey, tc.pageSize)
```

Reply:
Done.